src/segmentation.rs

```diff
@@ -0,0 +1,102 @@
+//! Segmenter trait and default implementations.
+//!
+//! Segmentation is how strings are split into tokens for comparison.
+//! For example, two strings that *appear* identical might have different byte-level representations.
```

Comment:
```suggestion
//! For example, two strings that *visually appear* identical might have different byte-level representations.
```
Perhaps pedantic, but also precise. 🙂